### PR TITLE
chore: implement arbitrary for `Config`

### LINKED ISSUE #2965: ConfigPartial shouldn't exist
> If it is wrong now (as the documentation tends to lag behind the code changes) then please modify it to reflect the current state. A wild TODO is, in general, not very helpful.

_Originally posted by @LesnyRumcajs in https://github.com/ChainSafe/forest/pull/2893#discussion_r1226458059_

### PR DESCRIPTION
**Pull request by lemmih**
## Summary of changes

Changes introduced in this pull request:

- lower `u64` and `usize` to `u32` such that they can be encoded as TOML.
- implement `Arbitrary` for `Config`.
- remove `ConfigPartial`.

## Reference issue to close (if applicable)

Closes #2965

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by LesnyRumcajs — APPROVED** [review]

**Review by jdjaustin — APPROVED** [review]

File: src/networks/mod.rs [hunk 1]
```diff
@@ -172,6 +173,21 @@ pub struct ChainConfig {
...
+impl quickcheck::Arbitrary for ChainConfig {
+    fn arbitrary(g: &mut quickcheck::Gen) -> ChainConfig {
+        use quickcheck::Arbitrary;
+        // Note: Not all `u64` values can be stored as TOML. Maybe we should use
+        // `u32` instead. Or maybe these values shouldn't be configurable at all.
+        ChainConfig {
+            network: Arbitrary::arbitrary(g),
+            genesis_cid: Arbitrary::arbitrary(g),
+            recent_state_roots: Arbitrary::arbitrary(g),
+            ..ChainConfig::default()
```
Comment (aatifsyed) [inline src/networks/mod.rs:186]: Why?

Comment (lemmih) [reply src/networks/mod.rs:186]: The config has to be serializable as TOML. But TOML does not support `u64`. Generating the `u64` fields will, correctly, cause our tests to fail. The note above talks about possible ways of fixing this.

Comment (aatifsyed) [reply src/networks/mod.rs:186]: I think you should fix this by doing `#[arbitrary(gen(|g| u32::arbitrary(g) as u64))]`

Comment (lemmih) [reply src/networks/mod.rs:186]: I'm not sure that's the best way forward. Ideally, bad states should be unrepresentable. Another option is to delete those fields entirely. I don't think we'll ever support arbitrary networks (ie, not calibnet, mainnet, or devnet).

**Review by aatifsyed — APPROVED** [review]

**Pull request merged**